AsyncDisplayKit/ASContextTransitioning.h

```diff
@@ -0,0 +1,36 @@
+//
+//  ASContextTransitioning.h
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 2/4/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import <AsyncDisplayKit/ASDisplayNode.h>
+
+@protocol ASContextTransitioning <NSObject>
+
+/**
+ @abstreact Defines if the given transition is animated
+ */
+- (BOOL)isAnimated;
+
+/**
+ @abstract The frame for the given node before the transition began.
+ @discussion Returns CGRectNull if the node was not in the hierarchy before the transition.
+ */
+- (CGRect)initialFrameForNode:(ASDisplayNode *)node;
+
+/**
+ @abstract The frame for the given node when the transition completes.
+ @discussion Returns CGRectNull if the node is no longer in the hierarchy after the transition.
+ */
+- (CGRect)finalFrameForNode:(ASDisplayNode *)node;
```

Comment:
These selector names are excellent.  Very in-line with the framework and awesomely short/clear compared to UIKit selectors.  So is the language of the documentation file-wide, apart from the slight ambiguity regarding CGRectNull